core/domain/blog_statistics_domain.py

```diff
@@ -0,0 +1,1165 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain objects relating to blog post statistics."""
+
+from __future__ import annotations
+
+import datetime
+
+from core import utils
+
+from typing import Dict, TypedDict, Union, overload
+
+# This is same as base_models.ID_Length.
+BLOG_POST_ID_LENGTH = 12
+
+
+@overload
+def _repack_views_stats(
+    stats: BlogPostViewsAggregatedStats,
+) -> None: ...
+
+
+@overload
+def _repack_views_stats(
+    stats: AuthorBlogPostViewsAggregatedStats,
+) -> None: ...
+
+
+def _repack_views_stats(
+    stats: Union[
+        BlogPostViewsAggregatedStats, AuthorBlogPostViewsAggregatedStats
+    ]
```

Comment:
It seems to me that you are using `Union[BlogPostViewsAggregatedStats, AuthorBlogPostViewsAggregatedStats]` type frequently. So I would suggest defining a type variable at the top of the file and use that instead of defining Union type again and again.